StorageUtility/Makefile

```diff
@@ -0,0 +1,98 @@
+# Compiler and tool definitions shared by all build targets 
+CC       = gcc
+CXX      = g++
+CFLAGS   = -pipe -O2 -fPIC -Wall -W -D_REENTRANT 
+CXXFLAGS = -pipe -std=c++14 -O2 -fPIC -Wall -W -D_REENTRANT 
+INCPATH  = -I. -I../Extern -I../Common -I../Common/Platform/x86x64 -I../Common/Platform/x86x64/Linux -I../vtStor/Platform/Linux -I../StorageUtility/Linux -I.
+CHK_DIR_EXISTS = test -d
+MKDIR   = mkdir -p
+
+# Define the target directories.
+OBJECTS_DIR         = ../Debug/StorageUtility
+TARGETDIR           = ../Debug
+COMMON_DIR          = ../Common
+GENERATEFILES_DIR   = ../GenerateFiles
+	
+# Target: libStorageUtility
+SOURCES = Scsi.cpp \
+	Ata.cpp \
+	-ILinux\Device.cpp \
+	-ILinux\StorageUtility.cpp
+	
+OBJECTS =  \
+	$(OBJECTS_DIR)/Ata.o \
+	$(OBJECTS_DIR)/Scsi.o \
+	$(OBJECTS_DIR)/Device.o \
+	$(OBJECTS_DIR)/StorageUtility.o
+	
+OBJECT_NAMES =  Ata.o \
+	Scsi.o \
+	Device.o \
+	StorageUtility.o
+
+TARGET  = libStorageUtility.a
+		
+all: generateFiles directories compile Makefile $(TARGETDIR)/libStorageUtility
+	
+generateFiles:
+	# output file                                                                           Templated file                                          xml file
+	python $(GENERATEFILES_DIR)/GenerateFiles.py $(COMMON_DIR)/ErrorCodes.h                 $(GENERATEFILES_DIR)/ErrorCodesTemplate.h               $(GENERATEFILES_DIR)/ErrorCodes.xml
+	python $(GENERATEFILES_DIR)/GenerateFiles.py $(COMMON_DIR)/ScanForHardwareChanges.h     $(GENERATEFILES_DIR)/ScanForHardwareChangesTemplate.h 	$(GENERATEFILES_DIR)/ScanForHardwareChanges.xml
+	python $(GENERATEFILES_DIR)/GenerateFiles.py $(COMMON_DIR)/BusType.h                    $(GENERATEFILES_DIR)/BusTypeTemplate.h                  $(GENERATEFILES_DIR)/BusType.xml
+	python $(GENERATEFILES_DIR)/GenerateFiles.py $(COMMON_DIR)/DeviceDataType.h             $(GENERATEFILES_DIR)/DeviceDataTypeTemplate.h           $(GENERATEFILES_DIR)/DeviceDataType.xml
+
+# Create the target directory (if needed)	
+directories:
+	@$(CHK_DIR_EXISTS) $(TARGETDIR) || $(MKDIR) $(TARGETDIR)
+	@$(CHK_DIR_EXISTS) $(OBJECTS_DIR) || $(MKDIR) $(OBJECTS_DIR)
+
+compile: Scsi.o Ata.o StorageUtility.o Device.o
+
+# Implicit rules
+.SUFFIXES: .o .c .cpp .cc .cxx .C
+
+.cpp.o:
+	$(CXX) -c $(CXXFLAGS) $(INCPATH) -o "$@" "$<"
+
+.cc.o:
+	$(CXX) -c $(CXXFLAGS) $(INCPATH) -o "$@" "$<"
+
+.cxx.o:
+	$(CXX) -c $(CXXFLAGS) $(INCPATH) -o "$@" "$<"
+
+.C.o:
+	$(CXX) -c $(CXXFLAGS) $(INCPATH) -o "$@" "$<"
+
+.c.o:
+	$(CC) -c $(CFLAGS) $(INCPATH) -o "$@" "$<"
+	
+
+# Compile source files into .o files
+Scsi.o: Scsi.cpp Scsi.h
```

Comment:
Use loop to create .o file from .cpp, If we have 100 cpp files we need 200 lines to do it